cclib/parser/gaussianparser.py

```diff
@@ -213,6 +214,233 @@ def extract(self, inputfile, line):
+            calc_type = 'ground_state'
+            calcfc_found, ts_found = False, False
+            for keyword in keywords_line.split():
+                if keyword.lower().find('opt') > -1:
+                    if keyword.lower().find('calcfc') > -1:
+                        calcfc_found = True
+                    if keyword.lower().find('ts') > -1:
+                        ts_found = True
+                elif keyword.lower().startswith('emp'):
+                    qm_disp = keyword
+            if calcfc_found and ts_found:
+                calc_type = 'transition_state'
+
+            self.metadata["dispersion model"] = qm_disp
+            self.metadata["calc_type"] = calc_type
+
+        # Extract grid type
+        # ExpMin= 9.52D-02 ExpMax= 1.36D+04 ExpMxC= 4.63D+02 IAcc=3 IRadAn=         5 AccDes= 0.00D+00
+        elif line[1:8] == 'ExpMin=':
+            grid_lookup = {1: 'sg1', 2: 'coarse', 4: 'fine', 5: 'ultrafine', 7: 'superfine'}
+            IRadAn = int(line.strip().split()[-3])
+            grid = grid_lookup[IRadAn]
+            self.metadata["grid"] = grid
+
+        # Extract <S**2> before and after spin annihilation
+        # <Sx>= 0.0000 <Sy>= 0.0000 <Sz>= 0.0000 <S**2>= 1.0237 S= 0.6286
+        # <L.S>= 0.000000000000E+00
+        # Annihilation of the first spin contaminant:
+        # S**2 before annihilation     1.0237,   after     0.2796
+        elif 'S**2 before annihilation' in line:
+            if not hasattr(self, 'spinexpect'):
+                self.set_attribute('spinexpect', [])
+                self.set_attribute('spinexpectanni', [])
+            self.append_attribute('spinexpectanni', float(line.strip().split()[-1]))
+            self.append_attribute('spinexpect', float(line.strip().split()[-3][:-1]))
+
+        # Extract symmetry number, rotational constants and rotational temperatures
+        elif 'Rotational symmetry number' in line:
+            symmno = int(line.strip().split()[3].split(".")[0])
+            self.set_attribute('rotsymno', symmno)
+
+        # Extract Rotational Constants
+        # Example:
+        # Rotational constants (GHZ):           3.13081     1.24272     0.88960
+        # OR for linear molecules:
+        # Rotational constants (GHZ): ************ 12.73690 12.73690
+        # Note: rotational constant will be converted to wavenumber units (1/cm) to standardize across parsers
+        elif line.find('Rotational constants (GHZ):') > -1:
+            # Replacing : with ' ' is to handle overflow cases like:
+            #  Rotational constants (GHZ):***************     11.1953814     11.1953785
+            split_line = line.strip().replace(':', ' ').split()
+            roconst = []
+            for value in split_line[-3:]:
+                try:
+                    roconst.append(float(value))
+                
+                except ValueError:
+                    roconst.append(numpy.nan)
+            
+            self.append_attribute("rotconsts", roconst)
```

Comment:
This change is to address #1270. It would normally be easier to have this as it's own PR but because it's  a small change and it's already bundled in this one I think we should just leave it for now.

The problem is the classic Fortran text overflow, which dumps a bunch of asterisks instead of the large number it was trying to print. Linear molecule are susceptible to this because their rotational constant in the linear axis is huge (and probably trending towards infinity?). There's actually a comment in the current code addressing this, but the parsing does not address it.

@jvalegre thanks for this (very pythonic!) initial fix. Sadly we can't just ignore one of the axis as you suggest, because if we parse multiple rotational constants (in an opt for example), some of them will have 3 values, and some only 2, and arrays don't work like this.

I've updated to store missing values as `nan`, but perhaps `0.0` or even `inf` might be better options?

We will also need a regression test for this fix.

Reply:
sounds good - could you use `nan`? That way it'd be easy for me to adapt GoodVibes (retrieve array from cclib, then remove `nan`)